apps/meteor/app/oauth2-server-config/server/admin/methods/addOAuthApp.ts

```diff
@@ -33,17 +41,39 @@ export async function addOAuthApp(application, uid) {
 		});
 	}
 
-	application.clientId = Random.id();
-	application.clientSecret = Random.secret();
-	application._createdAt = new Date();
-	application._createdBy = await Users.findOne(uid, { projection: { username: 1 } });
-	application._id = (await OAuthApps.insertOne(application)).insertedId;
-	return application;
+	const createdBy = await Users.findOne<Pick<IUser, '_id' | 'username'>>(uid, { projection: { username: 1 } });
+	if (!createdBy || !createdBy._id || !createdBy.username) {
+		throw new Meteor.Error('user-not-defined', 'User not defined.', {
+			method: 'addOAuthApp',
+		});
+	}
+
+	const inserted = {
+		...application,
+		clientId: Random.id(),
+		clientSecret: Random.secret(),
+		_createdAt: new Date(),
+		_updatedAt: new Date(),
+		_createdBy: {
+			_id: createdBy._id,
+			username: createdBy.username,
+		},
+	};
+
+	return {
+		...inserted,
+		_id: (await OAuthApps.insertOne(inserted)).insertedId,
+	};
```

Comment:
Should we combine these and return the object right away?

Reply:
I think we can't do that since the `_id` field depends on the data from the `inserted` object :thinking: